### PR TITLE
Baseline Model 

### PR DESCRIPTION
**Pull request by MostafaRahgouy**
We split the train set into two parts(train and test) which we assigned 80% to train set and 20% for testing.

### CODE REVIEW

**Comment by Rahgooy** [comment]
Thanks. It is great!
Please, run the evaluation with the train data and report.

File: baseline_model.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-tr', '--train', help="path to semi_train.txt dataset")
+    parser.add_argument('-ts', '--test', help="path to test.txt dataset")
+    parser.add_argument('-o', help="path to submition output input/res",default="input")
+    args = parser.parse_args()
+    if args.train is None or args.test is None:
+        parser.print_usage()
+        exit()
+    return args
```
Comment (Rahgooy) [inline baseline_model.py:15]: Consider using DocPie(http://docpie.comes.today/)

File: baseline_model.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            bio_freqs_word[word_stem] += (int(bio[0]) + int(bio[1]))
+            occurrences_word[word_stem] += 1
+    
+    
+    model = build_model(bio_freqs_word, occurrences_word)
+
+    print("testing the model ... ")
+    word_id_lsts, word_lsts, _, _, _, _  = rw.read_data(path_to_test)
+    predictions = predict(word_lsts, model)
+    rw.write_results(word_id_lsts, word_lsts, predictions, os.path.join(path_to_output, "submission.txt"))
```
Comment (Rahgooy) [inline baseline_model.py]: Create a class, named `WordFrequencyModel` with `fit` and `predict` functions and move these stuff into it.
Here you just call the `fit` function with the input and labels. and also the `predict` function with the input. This architecture is consistent with Scikit-learn models.
It would be nice to save the learned model using pickle in the `models` folder.

**Review by Rahgooy — COMMENTED** [review]
The code is very clear and well written. I had some suggestions for you!

**Comment by MostafaRahgouy** [comment]
> Thanks. It is great!
> Please, run the evaluation with the train data and report.

Please, visit the "Investigate the importance of word frequencies in predicting the bold words" issue to see the results.

File: baseline_model.py (hunk 2 above)
Comment (MostafaRahgouy) [reply baseline_model.py]: We changed the code as you requested. we added save and load methods to the class. Also, we save the learned model to the `models` folder as the `WordFrequencyModel.pkl` name.

File: baseline_model.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    
+    mkdir(path_to_output)
+    path_to_output = os.path.join(path_to_output , "res")
+    mkdir(path_to_output)
+
+    _, word_lsts, _, bio_lsts, _, _  = rw.read_data(path_to_train)
+
+    print("training the model ... ")    
+    wfm_model = wfm(word_lsts, bio_lsts)
+    wfm_model.fit()
```
Comment (Rahgooy) [inline baseline_model.py]: Sorry to bother you again. The inputs and labels should be the parameters of `fit` not the class constructor.

**Comment by Rahgooy** [comment]
Otherwise, the code looks perfect. Thanks.

File: baseline_model.py (hunk 3 above)
Comment (MostafaRahgouy) [reply baseline_model.py]: We did it. Is that you meant?

**Comment by MostafaRahgouy** [comment]
> Otherwise, the code looks perfect. Thanks.

good to hear that.

**Comment by Rahgooy** [comment]
Please move all the data files to the datasets directory. Also, create a folder for the results and move the results file(e.g. overall.md files) into it.

**Comment by MostafaRahgouy** [comment]
> Please move all the data files to the datasets directory. Also, create a folder for the results and move the results file(e.g. overall.md files) into it.

Do you have any suggestions for name of "semi_train"?

**Comment by Rahgooy** [comment]
> > Please move all the data files to the datasets directory. Also, create a folder for the results and move the results file(e.g. overall.md files) into it.
> 
> Do you have any suggestions for name of "semi_train"?

`train-test-split.txt`, and `train-train-split.txt`?!

**Comment by MostafaRahgouy** [comment]
> > > Please move all the data files to the datasets directory. Also, create a folder for the results and move the results file(e.g. overall.md files) into it.
> > 
> > 
> > Do you have any suggestions for name of "semi_train"?
> 
> `train-test-split.txt`, and `train-train-split.txt`?!

Thanks

**Comment by HamedBabaei** [comment]
> Please move all the data files to the datasets directory. Also, create a folder for the results and move the results file(e.g. overall.md files) into it.

We add the datasets to `datasets` dir and results to `results` dir as you want.

**Comment by Rahgooy** [comment]
Nice!
Thanks for the hard work.
Merged.
